### PR TITLE
Move Sha1ForNonSecretPurposes under Common

### PR DESCRIPTION
**Pull request by jkotas**
Avoid code duplication and contribute to #45237 and #45032

### CODE REVIEW

**Review by GrabYourPitchforks — APPROVED** [review]
Straightforward enough :)

**Comment by GrabYourPitchforks** [comment]
No harm in duplicating this line as an internal helper within this file.

https://github.com/dotnet/runtime/blob/fe80a77dac16de2451be60563123b8fbf8ea8355/src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs#L418-L419

**Comment by jkotas** [comment]
The OOB library has BitOperations stub for this and other reasons. It just was not in the right namespace to be picked up here.

**Comment by GrabYourPitchforks** [comment]
Looks like _System.Net.WebSockets.WebSocketProtocol.csproj_ still references _BitOperations.cs_? Removing the ifdefs could break that project.

**Comment by GrabYourPitchforks** [comment]
Unit test failure might be known issue https://github.com/dotnet/runtime/issues/24181?

File: src/libraries/Common/src/System/Sha1ForNonSecretPurposes.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            _w[84] = 0xC3D2E1F0;
+        }
+
+        /// <summary>
+        /// Adds an input byte to the hash.
+        /// </summary>
+        /// <param name="input">Data to include in the hash.</param>
+        public void Append(byte input)
+        {
+            _w[_pos / 4] = (_w[_pos / 4] << 8) | input;
```
Comment (marek-safar) [inline src/libraries/Common/src/System/Sha1ForNonSecretPurposes.cs:42]: nit: cache the index `_pos / 4` calculation

Comment (jkotas) [inline src/libraries/Common/src/System/Sha1ForNonSecretPurposes.cs:42]:
```suggestion
            int idx = _pos >> 2;
            _w[idx] = (_w[idx] << 8) | input;
```

**Pull request merged**